### PR TITLE
Base styles

### LINKED ISSUE #1: Button Styles
Need to define a base style for how our buttons will look, then need to add in contextual variants
E.g. 
Success button = green
Error button = red

### LINKED ISSUE #2: Typography
Need to define typography rules (fonts/sizes)

### LINKED ISSUE #3: UI Demo
There should be a page that allows the styling for every component/UI element to be easily visible

### LINKED ISSUE #8: Generalise Stylesheet
Currently the stylesheet is geared up specifically for BYBA colours. We need to generalise the spreadsheet to allow for other association colours.

This includes things like renaming button colours (e.g. "red" should be "danger")

Perhaps this could be achieved by defaulting to black/white and allowing for an extra stylesheet as an override?

### LINKED ISSUE #9: Build Tools
There should be a set of build tools that runs extra tasks on the CSS - adding vendor prefixes, injecting association variables, etc.

These build tools should be callable from the Bandmaster Admin Site so that the correct styles can be generated for each new site

### PR DESCRIPTION
**Pull request by rugulous**
There is a lot in here, soz! 

This pull request contains the following:

- Mixins/functions to allow for calculating the most accessible text colour for a given background (used to colour button text)
- Folders reorganised to follow an adapted version of the 7-1 Sass Architecture (last 2 removed as not relevant to a framework)
- 3 button styles (`default`, `error`, `success`) - this closes #1 
- styles for h1-h6, .h1-.h6, and general body text - this closes #2 
- `demo` directory containing a page to view all the included UI components - this closes #3 
- Default/overridable colours and fonts to decouple the stylesheet from general BYBA styles - this makes the stylesheet more generic and closes #8 
- `Builder` class that can be used to override said colours and fonts, and to write the resulting CSS to a specified file. Also a "test" script to demo functionality - these close #9 
- Updated README to match new folder structure and build task
- Neatening up of `style.scss` by creating `_index.scss` files for different directories - so the main entry point can literally just import each directory and let them sort themselves out 

Happy reading!

### CODE REVIEW

File: scss/vendors/_normalise.scss [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+    /* Grouping content
+       ========================================================================== */
+
+/**
+ * 1. Add the correct box sizing in Firefox.
+  * 2. Show the overflow in Edge and IE.
+   */
+
+hr {
+	  box-sizing: content-box; /* 1 */
```
Comment (matt-fidd) [inline scss/vendors/_normalise.scss]: Indentation in here looks a little wonky in places. Might be best downloading a copy as opposed to copying and pasting it.

File: scss/abstracts/_variables.scss [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+$bg: #EBEBEB !default;
+
+$colour-contrast-dark: $black;
+$colour-contrast-light: $white;
+$min-contrast-ratio: 4.5 !default;
+
+//put in an array so we can loop over contextual stylings etc
+$colours: (
+    "default": $default,
```
Comment (matt-fidd) [inline scss/abstracts/_variables.scss]: Indentation here is 4 spaces where in all other files it is tabs

File: scss/abstracts/mixins/_buttons.scss [hunk 3]
```diff
@@ -0,0 +1,34 @@
+//button mixins
+
+//style button based on passed colours
+
+@mixin button-variant(
+	$background,
+	$foreground: colour-contrast($background),
+	$hover_bg: color.scale($background, $lightness: -30%),
+	$hover_fg: $foreground
+
```
Comment (matt-fidd) [inline scss/abstracts/mixins/_buttons.scss]: Should we have a newline here? It doesn't appear in other mixin declarations.

**Review by matt-fidd — CHANGES_REQUESTED** [review]

**Comment by matt-fidd** [comment]
We could make this a module so that sass doesn't have to be installed on the host machine if we wanted? That way it would download it's own copy of sass when its installed.

**Comment by matt-fidd** [comment]
I've just tried cloning and using the `test.js` script and I'm getting.

```
$ node build/test.js 
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module 'sass'
Require stack:
- .../Style/build/Builder.js
- ...Style/build/test.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/data/byba/Style/build/Builder.js:1:14)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1155:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '.../Style/build/Builder.js',
    '.../Style/build/test.js'
  ]
}
```

File: scss/vendors/_normalise.scss (hunk 1 above)
Comment (rugulous) [reply scss/vendors/_normalise.scss]: You right. I've `curl`'d it now so any further issues are problems with the actual stylesheet!!!

**Comment by rugulous** [comment]
@matt-fidd how's this?

**Review by matt-fidd — APPROVED** [review]
Looks good!

**Pull request merged**